geode-core/src/main/java/org/apache/geode/management/internal/cli/UpdateAllConfigurationsByDefault.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.internal.cli;
+
+import org.apache.geode.cache.configuration.CacheConfig;
+
+/**
+ * A marker for {@link org.apache.geode.management.cli.SingleGfshCommand} classes that should
+ * iterate over multiple {@link org.apache.geode.cache.configuration.CacheConfig} objects
+ * when updating the {@link org.apache.geode.distributed.ConfigurationPersistenceService}.
+ */
+
+public interface UpdateAllConfigurationsByDefault {
+  /**
+   * A filter to determine which {@link org.apache.geode.cache.configuration.CacheConfig} objects
+   * should be updated by
+   * {@link org.apache.geode.management.cli.SingleGfshCommand#updateClusterConfig(String, org.apache.geode.cache.configuration.CacheConfig, Object)}.
+   *
+   * Typical use will be to examine the configuration to verify the change that will be applied by
+   * {@link org.apache.geode.management.cli.SingleGfshCommand#updateClusterConfig(String, org.apache.geode.cache.configuration.CacheConfig, Object)}
+   * will be valid.
+   *
+   * @param group Name of the group for the given CacheConfig.
+   * @param config The persisted configuration object. Can be null.
+   * @param configObject The configuration object set by the gfsh command
+   * @return true if this configuration should be updated.
+   */
+  boolean configurationFilter(String group, CacheConfig config, Object configObject);
```

Comment:
I thought we only want to add a marker interface, if a command implement this method, it will update all configurations instead of just one of it.

Reply:
I went this route to avoid (a) a counter-intuitive call stack and (b) confusing user experience.

* If the `updateClusterConfig` call reached out to every config, it kind of violates the intent of the method's signature.  That method is called in the `CommandExecutor` as a lambda, applied to the configuration of the specified group.  To achieve the same effect, we would need the `updateClusterConfig` to (a) handle as a special case when the `String group = "cluster"` vs when that `group` is an actual member group, and in that special case it would need to call itself recursively.  All this from within a lambda that appears to only update the `"cluster"` config, from the point of view of its use in `CommandExecutor`, effectively as `updateClusterConfig("cluster", config, object)`.  In addition to greatly overstepping the expected bounds of that call, this suffers an additional problem mentioned below.
* If a command implementing this interface is applied to every possible group, the `updateClusterConfig` must itself apply some filtering to avoid, e.g., attempting to remove an async event queue that this particular config does not contain.  That would be fine, except...
* We get the message `Changes to configuration for group 'XYZ' are persisted` or `failed to update cluster config for XYZ` for every group that we attempt to update.  The result of this is that we might have the pseudo-sequence

```
gfsh> create async-event-queue --id=aeqA --group=groupA
gfsh> create something else --group=groupB
gfsh> create another something --group=groupC
gfsh> destroy async-event-queue --id=aeqA # implicitly --group="cluster"
[...]
Changes to configuration for group 'cluster' are persisted.
Changes to configuration for group 'groupA' are persisted.
Changes to configuration for group 'groupB' are persisted.
Changes to configuration for group 'groupC' are persisted.
```

As a user, it would be confusing to see a litany of messages regarding updates to my persisted configuration, considering that it only truthfully changed one of them.

And to return to the previous concern, if `updateClusterConfig` reached out to every group in turn, we would only see the message `Changes to configuration for group 'cluster' are persisted`, despite the fact that (a) no changes to that group actually occurred, and (b) changes to another group ***did*** occur.

So, while a filter on this interface might seem like undue ceremony, I believe that it is necessary to avoid convoluted, recursive calls to update cluster config, as well as being sensible to an end-user what updates have actually occurred.

I don't see a better, ready solution that doesn't severely violate separation of concerns between the command, configuration, and `CommandExecutor`.

Reply:
Examining it now, however, I see that I did not update the `DestroyAsyncEventQueueCommandDUnitTest` to codify the limited scope of the `Changes to configuration [...]` messages.  Those tests that do not specify `--group` should be updated to ensure only the relevant messages are displayed.

Reply:
I'm confused about your comment about the possible need for recursion. I thought that the simple presence of this (marker) interface would result in `updateClusterConfig` being iterated over with all groups.

One overall addition would be to have `updateClusterConfig` return a boolean indicating whether CC was actually updated. That would allow for more accurate user messages.